oak-query-spi/src/main/java/org/apache/jackrabbit/oak/spi/query/QueryLimits.java

```diff
@@ -39,12 +39,22 @@ default boolean getImprovedIsNullCost() {
         return true;
     }
 
+    /**
+     * See OAK-11214. This method is used for backward compatibility
+     * (bug compatibility) only.
+     *
+     * @return true, except when backward compatibility for OAK-11214 is enabled
+     */
+    default public boolean getOptimizeFunctionInList() {
```

Comment:
to improve clarity and readability, can we rename this with something like:

* `isOptimizeFunctionInListEnabled`
* `isFunctionInListOptimized`
* `useOptimizedFunctionInList`

Reply:
Yes, it looks better.

On this point:

> And I hope we can remove the toggle in about a year or so.

the article below has an interesting section on `Managing the carrying cost of Feature Toggles`:

> Savvy teams view the Feature Toggles in their codebase as inventory which comes with a carrying cost and seek to keep that inventory as low as possible. In order to keep the number of feature flags manageable a team must be proactive in removing feature flags that are no longer needed. Some teams have a rule of always adding a toggle removal task onto the team's backlog whenever a Release Toggle is first introduced. Other teams put “expiration dates” on their toggles. Some go as far as creating “time bombs” which will fail a test (or even refuse to start an application!) if a feature flag is still around after its expiration date.

https://martinfowler.com/articles/feature-toggles.html#WorkingWithFeature-flaggedSystems